src/Pdf.jsx

```diff
@@ -252,8 +252,13 @@ class Pdf extends React.Component {
     if (page) {
       const { canvas } = this;
       const canvasContext = canvas.getContext('2d');
+      let DPIScale = 1;
+      DPIScale = window.devicePixelRatio;
       const { scale, rotate } = this.props;
-      const viewport = page.getViewport(scale, rotate);
+      const adjustedScale = scale * DPIScale;
+      const viewport = page.getViewport(adjustedScale, rotate);
+      canvas.style.width = `${viewport.width / DPIScale}px`;
```

Comment:
Both need to be adjusted in order for it to work. The [article](https://www.html5rocks.com/en/tutorials/canvas/hidpi/) adjusts both as well:

```` js
        canvas.width = oldWidth * ratio;
        canvas.height = oldHeight * ratio;

        canvas.style.width = oldWidth + 'px';
        canvas.style.height = oldHeight + 'px';
````